be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -552,6 +575,58 @@ Status BetaRowsetWriter::_segcompaction_ramaining_if_necessary() {
     return status;
 }
 
+Status BetaRowsetWriter::_segcompaction_finalpass_wait() {
+    Status status = Status::OK();
+    DCHECK_EQ(_is_doing_segcompaction, false);
+    if (!config::enable_storage_vectorization) {
+        return Status::OK();
+    }
+    if (_segcompaction_status.load() != OLAP_SUCCESS) {
+        return Status::OLAPInternalError(OLAP_ERR_SEGCOMPACTION_FAILED);
+    }
+
+    int64_t num_seg = _is_segcompacted() ? _num_segcompacted : _num_segment;
+    if (num_seg <= FINAL_SEGCOMPACTION_NUM_THRESHOLD) {
+        LOG(INFO) << "num_seg less than max_segment_num_per_rowset, skip final pass: " << num_seg
```

Comment:
How about the case that the data size is quite big, and `total_size/MAX_SEGMENT_SIZE > FINAL_SEGCOMPACTION_NUM_THRESHOLD`?